app/ui/legacy/src/main/java/com/fsck/k9/activity/AccountList.kt

```diff
@@ -0,0 +1,151 @@
+package com.fsck.k9.activity
+
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemClickListener
+import android.widget.ArrayAdapter
+import com.fsck.k9.Account
+import com.fsck.k9.BaseAccount
+import com.fsck.k9.K9.isShowUnifiedInbox
+import com.fsck.k9.Preferences.Companion.getPreferences
+import com.fsck.k9.search.SearchAccount.Companion.createUnifiedInboxAccount
+import com.fsck.k9.ui.R
+import com.google.android.material.textview.MaterialTextView
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * Activity displaying the list of accounts.
+ *
+ * Classes extending this abstract class have to provide an [.onAccountSelected]
+ * method to perform an action when an account is selected.
+ */
+abstract class AccountList : K9ListActivity(), OnItemClickListener {
+
+    private val scope = MainScope()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setResult(RESULT_CANCELED)
+        setLayout(R.layout.account_list)
+
+        listView.apply {
+            onItemClickListener = this@AccountList
+            itemsCanFocus = false
+        }
+    }
+
+    /**
+     * Reload list of accounts when this activity is resumed.
+     */
+    public override fun onResume() {
+        super.onResume()
+        loadAccounts()
+    }
+
+    /**
+     * Cancel all running coroutines when this activity is paused.
+     */
+    public override fun onPause() {
+        super.onPause()
+        scope.cancel()
+    }
```

Comment:
The manual coroutine scope management can be avoided by adding the following to `onCreate()`:

```kotlin
lifecycleScope.launch {
    repeatOnLifecycle(Lifecycle.State.RESUMED) {
        loadAccounts()
    }
}
```

together with:

```kotlin
private suspend fun loadAccounts() {
    val accounts = withContext(Dispatchers.IO) {
        Preferences.getPreferences().getAccounts()
    }

    populateListView(accounts)
}
```

Reply:
I personally don't like the Lifecycle APIs, but it seems to simplify the code.